### PR TITLE
Internal: Github action of creating release branches [ED-10633]

### PR DESCRIPTION
**Pull request by maork-elementor**

### CODE REVIEW

File: .github/workflows/create-release-branch.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref }}
+      - name: Create Branch
+        run: |
+          git config --global user.email "ElementorBot@elementor.com"
+          git config --global user.name "ElementorBot"
```
Comment (matipojo) [inline .github/workflows/create-release-branch.yml:18]: - Move to before the commit action.
- 
```suggestion
          git config --global user.name "${{ secrets.MAINTAIN_USERNAME }}"
          git config --global user.email "${{ secrets.MAINTAIN_EMAIL }}"
```

File: .github/workflows/create-release-branch.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          echo "Current branch version is $current_branch_version"
+
+          # e.g 3.14
+          if ! [[ $current_branch_version =~ ^[0-9]+\.[0-9]+$ ]]; then
+            echo "Invalid version ${{ github.ref }}"
+            exit 1
+          fi
+
+          # e.g 3.15
+          increment_version=$(echo $current_branch_version | awk -F. -v OFS=. '{$NF++;print}')
```
Comment (matipojo) [inline .github/workflows/create-release-branch.yml:31]: Test various cases

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/workflows/create-release-branch.yml (hunk 1 above)
Comment (maork-elementor) [reply .github/workflows/create-release-branch.yml:18]: Done

File: .github/workflows/create-release-branch.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Create Branch
+        run: |
+          git config --global user.email "ElementorBot@elementor.com"
+          git config --global user.name "ElementorBot"
+
+          #e.g refs/heads/3.14
+          current_branch_version=$(echo ${{ github.ref }} | cut -d'/' -f 3)
+          echo "Current branch version is $current_branch_version"
+
+          # e.g 3.14
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/create-release-branch.yml:24]: Done

File: .github/workflows/create-release-branch.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref }}
+      - name: Create Branch
+        run: |
+          git config --global user.email "ElementorBot@elementor.com"
+          git config --global user.name "ElementorBot"
+
+          #e.g refs/heads/3.14
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/create-release-branch.yml:20]: Done

File: .github/workflows/create-release-branch.yml (hunk 2 above)
Comment (maork-elementor) [reply .github/workflows/create-release-branch.yml:31]: Work on case such as 

3.13
3.0
3.6

File: .github/workflows/create-release-branch.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+name: Create Release Branch
+# Create a release branch from the current release branch ref 
+
+on:
+  workflow_dispatch:
+
+jobs:
+  create-branch:
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/create-release-branch.yml:8]: Done

**Review by matipojo — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**